common/SigUtil.cpp

```diff
@@ -38,57 +38,63 @@
 #include "Common.hpp"
 #include "Log.hpp"
 
+namespace
+{
 #ifndef IOS
-static std::atomic<bool> TerminationFlag(false);
-static std::atomic<bool> ShutdownRequestFlag(false);
+
+/// The valid states of the process.
+enum class RunState : char
+{
+    Run = 0, //< Normal up-and-running state.
+    ShutDown, //< Request to shut down gracefully.
+    Terminate //< Immediate termination.
+};
+
+/// Single flag to control the current run state.
+static std::atomic<RunState> RunStateFlag(RunState::Run);
```

Comment:
Note that you're inside an anonymous namespace here, so a 'static' is redundant. core.git CI even errors out in such cases.